app/services/data_exports/data_storage.rb

```diff
@@ -0,0 +1,125 @@
+require 'fileutils'
+require 'securerandom'
+
+module DataExports
+  class NoStoredFilesFoundError < StandardError; end
+
+  class DataStorage
+    # Public: downloads all transcription files for a given transcription
+    # returns path to zip file
+    def zip_transcription_files(transcription)
+      Dir.mktmpdir { |directory_path|
+        if transcription.files.attached?
+          transcription_folder = download_transcription_files(transcription, directory_path)
+          yield zip_files(directory_path, transcription_folder)
+        else
+          raise NoStoredFilesFoundError.new("No stored files found for transcription with id '#{transcription.id}'")
+        end
+      }
+    end
+
+    # Public : downloads all transcription group files for a given group
+    # returns path to zip file
+    def zip_group_files(transcriptions)
+      Dir.mktmpdir { |directory_path|
+        group_folder = File.join(directory_path, "group_#{transcriptions.first.group_id}")
+        FileUtils.mkdir_p(group_folder)
+
+        metadata_file_gen = AggregateMetadataFileGenerator.new(group_folder)
+        metadata_file_gen.generate_group_file(transcriptions)
+
+        transcriptions.each do |t|
+          download_transcription_files(t, group_folder) if t.files.attached?
+        end
+
+        yield zip_files(directory_path, group_folder)
+      }
+    end
+
+    # Public : downloads all files for a given workflow
+    # returns path to zip file
+    def zip_workflow_files(workflow)
+      Dir.mktmpdir { |directory_path|
```

Comment:
multi line blocks should be `do end` not `{ }` 

I'd say install rubocop linter in your IDE or dev env an use this, it'll pick these style errors etc up as you work and guide you to clearer more readable code

Reply:
Yeah, apparently I had disabled rubocop within vs code for some reason that I can't remember, but Ive got it back in action now, it will _show me the way_ ✨